backends/p4tools/modules/smith/common/statements.cpp

```diff
@@ -388,4 +404,88 @@ IR::ReturnStatement *StatementGenerator::genReturnStatement(const IR::Type *tp)
     return new IR::ReturnStatement(expr);
 }
 
+/// Generate a for-loop statement.
+IR::ForStatement *StatementGenerator::genForLoopStatement(bool is_in_func) {
+    std::string loopVar = generateLoopControlVariable();
+    // TODO(zzmic): Determine the exact range of the bit field width and the upper
+    // bound (keep in mind that buffer overflow should be avoided).
+    int bitFieldWidth = Utils::getRandInt(1, 64);
+    // Prevent the potential buffer overflow issue.
+    // Fow now, set the upper bound to 100.
+    long long maxValue =
+        (bitFieldWidth < 64) ? ((1LL << bitFieldWidth) - 1) : (std::numeric_limits<int64_t>::max());
+    long long upperBound = Utils::getRandInt(1, std::min(100LL, maxValue));
+
+    // Debugging statements
+    std::cout << "genForLoopStatement - bitFieldWidth: " << bitFieldWidth
+              << ", maxValue: " << maxValue << ", upperBound: " << upperBound << std::endl;
+
+    // Create the IR nodes for the for-loop components.
+    auto *initExpr =
+        new IR::Declaration_Variable(IR::ID(loopVar), IR::Type_Bits::get(bitFieldWidth),
+                                     new IR::Constant(IR::Type_Bits::get(bitFieldWidth), 0));
+    auto *condExpr = new IR::Lss(IR::Type_Boolean::get(), new IR::PathExpression(loopVar),
+                                 new IR::Constant(IR::Type_Bits::get(bitFieldWidth), upperBound));
+    auto *updateStmt = new IR::AssignmentStatement(
+        new IR::PathExpression(loopVar),
+        new IR::Add(IR::Type_Bits::get(bitFieldWidth), new IR::PathExpression(loopVar),
+                    new IR::Constant(IR::Type_Bits::get(bitFieldWidth), 1)));
+    auto *bodyStmt = genBlockStatement(is_in_func);
+    // Fill `initExpr` and `updateStmt` into their corresponding indexed vectors.
+    // This is necessary due to the constructor defintions.
+    IR::IndexedVector<IR::StatOrDecl> initExprs;
+    IR::IndexedVector<IR::StatOrDecl> updateStmts;
+    initExprs.push_back(initExpr);
```

Comment:
I would just initialize the vector here instead of using push_back.

Reply:
> I would just initialize the vector here instead of using push_back.

Change applied.